sale_order_type_sale_journal/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Lorenzo Battistini - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.onchange('type_id')
+    def onchange_type_id(self):
```

Comment:
FYI, instead of calling super, you can also change the name of method to `onchange_type_id_sale_journal` for example and both methods will be called.